Enhance process information in `edmProvDump` output and include hardware information there

makortel opened the pull request:
#### PR description:

This PR is a follow-up to https://github.com/cms-sw/cmssw/pull/47355 to include the hardware information in "a nice way" in the `edmProvDump` output. The "nice way" required to overhaul the way the process information is printed out, that should now be more self-documenting and consistent with the other sections.

The last commit updates the `hltInfo` script that is known to depend on `edmConfigDump` output (thanks for adding the test in https://github.com/cms-sw/cmssw/pull/47860)

Context https://github.com/cms-sw/cmssw/issues/30044

Resolves https://github.com/cms-sw/framework-team/issues/1239

#### PR validation:

Unit tests pass

makortel committed:
Rename TestEdmConfigDump test as TestEdmProvDump

The test is really about edmProvDump

makortel committed:
Overhaul process information printing and extend testing

makortel committed:
Move event filtering dump below processes

makortel committed:
Update hltInfo script for edmProvDump output changes

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45524

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45524/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45524/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45525

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 40KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47249/summary.html
**COMMIT**: 1ae8312f2143283536fcfb16ac1e74669b7e5416
**CMSSW**: CMSSW_15_1_X_2025-07-16-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48562/47249/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-16-1100+931591/69398/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076521
 - DQMHistoTests: [Total failures: 62](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-16-1100+931591/69398/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076439
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-16-1100+931591/69398/triggerResults): no differences found

makortel committed:
Add an option to print the process history and configuration IDs

makortel commented:
The `ProcessConfiguration` elements of `ProcessHistoryVector` are apparently sorted according to the `ProcessConfigurationID` in
https://github.com/cms-sw/cmssw/blob/9fb8a43ab59a31febb604538ac864825162c53b3/IOPool/Provenance/src/CommonProvenanceFiller.cc#L32-L43

cmsbuild commented:
Pull request #48562 was updated. @Dr15Jones, @Martin-Grunewald, @cmsbuild, @makortel, @mmusich, @smuzaffar can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 36KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47368/summary.html
**COMMIT**: 74acc73da4c29280ed96aac35ad6b126f04b53ba
**CMSSW**: CMSSW_15_1_X_2025-07-23-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47368/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-23-1100+931591/69505/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-23-1100+931591/69505/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076481
 - DQMHistoTests: [Total failures: 31](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-23-1100+931591/69505/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076430
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-23-1100+931591/69505/triggerResults): no differences found

makortel commented:
@cmsbuild, please test

Let's see if this would shed some light to the test failure

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45603

cmsbuild commented:
Pull request #48562 was updated. @Dr15Jones, @Martin-Grunewald, @makortel, @mmusich, @smuzaffar can you please check and sign again.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 16KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47391/summary.html
**COMMIT**: c949aead2543afdeca14a8f51b64352a39a6d679
**CMSSW**: CMSSW_15_1_X_2025-07-25-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47391/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 0 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-25-1100+931591/69535/validateJR.html)
 - DQMHistoTests: Total files compared: 1
 - DQMHistoTests: Total histograms compared: 0
 - DQMHistoTests: [Total failures: 0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-25-1100+931591/69535/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 0
 - DQMHistoTests: Total skipped: 0
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0 KiB( 0 files compared)
 - Checked 0 log files, 0 edm output root files, 1 DQM output files

makortel commented:
test parameters:

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45608

makortel commented:
@cmsbuild, please test

makortel committed:
Sort the process histories according to the reduced id

To have the order stable, independent of the hardware of the node where the file was produced.

makortel commented:
@cmsbuild, please test

Hopefully sorting the process histories according to reduced ID (that is independent of the hardware) would solve the test failure.

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47481/summary.html
**COMMIT**: dc7e1a3eebc4b3810db71cce219a455e99ab2fde
**CMSSW**: CMSSW_15_1_X_2025-07-31-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/48562/47481/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-31-1100+931591/69629/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076122
 - DQMHistoTests: [Total failures: 14](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-31-1100+931591/69629/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076088
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-07-31-1100+931591/69629/triggerResults): no differences found

cmsbuild commented:
Pull request #48562 was updated. @Dr15Jones, @Martin-Grunewald, @cmsbuild, @makortel, @mmusich, @smuzaffar can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
Pull request #48562 was updated. @Dr15Jones, @Martin-Grunewald, @cmsbuild, @makortel, @mmusich, @smuzaffar can you please check and sign again.

makortel commented:
@cmsbuild, please test

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 28KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47516/summary.html
**COMMIT**: 22bc76fc87c79bc527a2cd885ea018169d908008
**CMSSW**: CMSSW_15_1_X_2025-08-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47516/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69663/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 7 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69663/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076122
 - DQMHistoTests: [Total failures: 34](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69663/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076068
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69663/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45686

cmsbuild commented:
-1

**Failed Tests**: UnitTests 
**Size**: This PR adds an extra 36KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47518/summary.html
**COMMIT**: 19dc2735a896acd653d2f2ed5a8ec067a6f273bf
**CMSSW**: CMSSW_15_1_X_2025-08-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47518/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Unit Tests

I found 1 errors in the following unit tests:

<pre>---> test TestEdmProvDump had ERRORS
</pre>

## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69666/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 3 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69666/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076122
 - DQMHistoTests: [Total failures: 23](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69666/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076079
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69666/triggerResults): no differences found

makortel commented:
Forgot to update the reference `edmProvDump` output files when I fixed the sorting

cmsbuild commented:
Pull request #48562 was updated. @Dr15Jones, @Martin-Grunewald, @cmsbuild, @makortel, @mmusich, @smuzaffar can you please check and sign again.

cmsbuild commented:
+1

**Size**: This PR adds an extra 32KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47522/summary.html
**COMMIT**: c57abded71cdc5089fdb678421a5e98167c66c1b
**CMSSW**: CMSSW_15_1_X_2025-08-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47522/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 2 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69673/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 7 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69673/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076122
 - DQMHistoTests: [Total failures: 77](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69673/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076025
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-04-1100+931591/69673/triggerResults): no differences found

Dr15Jones commented on IOPool/Common/bin/EdmProvDump.cc:
```diff
@@ -494,24 +326,43 @@ class ProvenanceDumper {
-  bool const printHardwareResourcesDescription_;
+  bool const sort_;
+
+  using ProcessSimpleIDsType = std::map<edm::ProcessConfigurationID, unsigned int>;
```
How about a comment about what the unsigned int is and how it is used?

makortel replied:
Added a comment

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-48562/45697

makortel commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Size**: This PR adds an extra 32KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-931591/47544/summary.html
**COMMIT**: 1e9cfc0c94eb7741afb30eff10679008d35b3997
**CMSSW**: CMSSW_15_1_X_2025-08-05-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/48562/47544/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-05-1100+931591/69695/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 5 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-05-1100+931591/69695/validateJR.html)
 - DQMHistoTests: Total files compared: 50
 - DQMHistoTests: Total histograms compared: 4076122
 - DQMHistoTests: [Total failures: 32](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-05-1100+931591/69695/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 4076070
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
 - Checked 215 log files, 184 edm output root files, 50 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_15_1_X_2025-08-05-1100+931591/69695/triggerResults): no differences found

makortel commented:
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/47071 and https://github.com/cms-sw/cmssw/issues/48504

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @antoniovilela, @sextonkennedy, @rappoccio, @mandrenguyen (and backports should be raised in the release meeting by the corresponding L2)
